chemist-aws/src/main/scala/AwsDiscovery.scala

```diff
@@ -314,16 +325,19 @@ class AwsDiscovery(
 
     val datacenter = in.getPlacement.getAvailabilityZone
 
-    val mirrorTemplate: Option[String] =
-      machineTags.get(AwsTagKeys.mirrorTemplate) orElse defaultInstanceTemplate
+    //Note: if node is not labeled explicitly we ignore it to avoid wasting time trying to continiously
+    // reconnect to incompatible instances
+    val mirrorTemplate: Option[String] = machineTags.get(AwsTagKeys.mirrorTemplate)
 
-    for {
+    val r = for {
       a <- toLocation(dns, datacenter, mirrorTemplate, Mirroring)
-      // _  = log.debug(s"discovered mirrioring template '$a'")
+      // _  = log.debug(s"discovered mirroring template '$a'")
     } yield AwsInstance(
       id = in.getInstanceId,
       tags = machineTags,
       locations = NonEmptyList(a)
     )
+
+    r.leftMap(message => (in.getInstanceId, message))
```

Comment:
or since this is the only place that toLocation is used the whole for comprehension + leftMap could just get wrapped into toLocation (which should then probably get renamed).